Test ivreg

jzangela opened the pull request:
Renamed the test_ivreg_ivreg.R file and changed its path.

jepusto reviewed (changes requested):
Thanks for adding these unit tests. I made a few further changes, so be sure to pull the latest from Github before adding further work of your own. 

Here are a few follow-up tasks to prepare this addition to be pulled into the main branch:
- It looks like `vcovCR()` returns some result for `ivreg` models, regardless of the estimation method used. However, based on [this](https://john-d-fox.github.io/ivreg/articles/Diagnostics-for-2SLS-Regression.html#robust-two-stage-instrumental-variables-regression-1), I don't think that the results are correct for `method = 'M'` or `method = 'MM'`. Please add a check to the function `vcovCR.ivreg()` that passes if `method = 'OLS'` but returns an error message otherwise. The error message could be something like "clubSandwich does not currently support ivreg models estimated using `method = 'M'` or `method = 'MM'`." See the function `vcovCR.lmerMod()` for an example of how to implement a check like this.
- Add some unit tests to `tests/testthat/test_ivreg_ivreg.R` to verify that `vcovCR.ivreg()` returns an error under the conditions described above. 
- Update the Description field in the `DESCRIPTION` file to indicate that clubSandwich now supports `ivreg` models from the ivreg package, when estimated by OLS.
- Update the Suggests field in the `DESCRIPTION` file to add the `ivreg` package.
- Edit the documentation in `R/ivreg.R` to indicate that clubSandwich only works with ivreg models estimated using OLS.
- Edit the documentation in `R/ivreg.R` to include an example using the `ivreg` package (it can be the exact same model as the current example that uses `AER::ivreg()`, or feel free to do a different model). Be sure to recompile the documentation when you've finished.

jzangela commented:
Hi James, 

I pushed some changes as discussed. Below is a brief summary.

1. Added a check in the `ivreg.R` file to the function `vcovCR.ivreg()` that passes if `method = 'OLS'`but returns an error if `method = 'M' or 'MM'`. The error message is set up as "clubSandwich does not currently support ivreg models estimated using method = 'M' or method = 'MM'."
2. Edited the documentation in the `ivreg.R` file to indicate that `clubSandwich` only works with `ivreg` models estimated using OLS.
3. Edited the documentation in the `ivreg.R` file to include an example using the `ivreg` package with data from `AER` package.
4. Add a unit test in the `test_ivgre_ivreg.R` file to verify that `vcovCR.ivreg()` returns an error.
5. Edited the `vcovCR.ivreg.Rd` documentation to indicate that `clubSandwich` only works with `ivreg` models estimated using OLS.
6. Updated the `Description` field in the `DESCRIPTION` file to indicate that `clubSandwich` now supports `ivreg` models from the `ivreg` package, when estimated by OLS.
7. Updated the `Suggests` field in the `DESCRIPTION` file to add the `ivreg` package.

Please have a look and let me know if we are good to move forward or if we need any modifications.

Thanks!

Best regards,
Jingru

jepusto reviewed (approved):
Looks good!
